custom_components/freeair_connect/FreeAir/__init__.py

```diff
@@ -481,12 +481,21 @@ def _parse(self, encrypted_data, timestamp, version, version_fa100):
         # encrypted_data = urllib.parse.unquote(encrypted_data) # this is probably not needed!
         encrypted_data = base64.b64decode(encrypted_data)
 
+        version_numbers = version.split("x")
+        if version_numbers[0] == 2 and (version_numbers[1] <= 13 or version_numbers[1] == 20 or version_numbers[1] == 21):
+            iv = "000102030405060708090a0b0c0d0e0f"
+            size = 16
+        else:
+            iv = "30313233343536373839303132333435"
+            size = 32
+
         # prepare initialization vector
-        iv = "000102030405060708090a0b0c0d0e0f"
         iv = binascii.unhexlify(iv)
+        
+        size = 32 # key size in bytes
```

Comment:
@mampfes overriding the size here is probably unintended? I don't think it will work for FW <= 2.13/2.20/2.21, but cannot test myself.

Reply:
Correct, this has been fixed already in the latest release.